packages/core/src/__tests__/refs.test.tsx

```diff
@@ -190,15 +181,11 @@ it('should manage subscription to stores referenced by refs', () => {
         }
       }
     },
+    sendEvent: jest.fn(),
   }
 
   const element = render(
-    <Template
-      components={BUILT_IN_COMPONENTS}
-      dataSource={dataSource}
-      onEvent={jest.fn()}
```

Comment:
hm I will miss having onEvent directly on the template component. It feels nice that this can be separated from the data source.

Reply:
Well, my initial thought process was that: Template is meant to be used with Data Source, because Template component has a tightly coupled relation with it (subscribing to stores and listening to changes).

If you don't want that Data Source-specific functionality, you can just use BaseTemplate directly which lets you set onEvent yourself as it's not really coupled to the implementation.